Foundation of New Governance Model (#98)
jupyter/governance: decision_making.md

# Decision-Making Guide

This document describes how Jupyter’s governing bodies make decisions. Because individuals often work across multiple decision-making groups, all Jupyter governing bodies are expected to follow this approach.

We seek to honor the principles of collaboration, inclusive participation, and responsive decision making. Some aspects of this decision-making guide are required of all teams. Others are provided as recommendations and are optional. We have clearly noted the optional aspects of decision making below.

> Finally, the Jupyter Board of Directors may either intervene, or be called upon by members of a team, in the event that issues arise in the decision-making process. This includes, but is not limited to: process ambiguity, violations of the decision making process, mitigating circumstances that require process exceptions, etc.

Reviewer: ```suggestion
Finally, the Jupyter Board of Directors may either intervene, or be called upon by members of a team, in the event that issues arise in the decision-making process. This includes, but is not limited to: process ambiguity, violations of the decision-making process, mitigating circumstances that require process exceptions, etc.
```

**Calling a vote.** When the discussion matures during the consensus-seeking phase, any member of the decision-making body can call the matter to a vote. When that member (the sponsor) calls the vote, they shall summarize the proposal in its current form for the entire decision-making body. After the proposal is seconded by another member of the decision-making body, members have seven days to vote. The governing body may consider longer voting periods as necessary for special circumstances, or _shorter periods only if all voting members are present_. The decision will be determined by a simple majority of non-blank votes for binary decisions (i.e., approving a proposal) and ranked choice for multi-class decisions (one among many, or several among many). The sponsor may update the proposal at any point during the voting period, in which case the voting period will be reset.

> **Voter participation and quorum.** All members of a decision-making body are required to participate in at least 2/3 of formal votes of that decision-making body per calendar year (teams can decide how to account for the specifics of this in low-activity projects, etc.). Members that have not met the 2/3 vote participation threshold for a year will automatically be asked to step down at the end of that year. Those individuals remain eligible to rejoin the decision-making body in the future as they become available to participate at the required level. The quorum for all formal votes will be 50% and a "blank" option will always be included, with the "blank" option counting towards the quorum but not included in totals for calculating results.

Reviewer 2: This policy is likely to negatively affect anyone who cannot, for various reasons, participate in all of the votes. Such people tend to come from less-privileged, more marginalized, and underrepresented groups. Merely by observing and speaking, members of underrepresented groups can make a immensely positive difference. These individuals should be able to remain at the table, if only so that they may bear witness to the decision-making process.

Reviewer 3: M, that's a reasonable point. IIRC, the metric for participation is to encourage engaged individuals in a decision making process. Onboarding and offboarding processes are important for healthy governance. Oftentimes, someone feels guilty about stepping down when too busy. 

I believe if we soften the wording to better reflect that we want engaged individuals but recognize that sometimes life happens: 

~~Members that have not met the 2/3 vote participation threshold for a year will automatically be asked to step down at the end of that year.~~
Members who miss 1/3 of the votes will be asked if they would like to continue to serve the following year. If the individual chooses to step down, ~~Those individuals~~ the individual remains eligible to rejoin the decision-making body in the future.

Reviewer 2: Agreed that offboarding processes are important! That feels something that should be available to all members (regardless of their participation in votes).

To make this available to all members, what if _every_ member were encouraged to yearly reaffirm that they wish to remain on the decision making body (with an off-boarding procedure then made available for any member who wishes to no longer work on the project).

Some of the other language here feels like it needs to be softened to match your suggestion’s tone. Drawing from your description of the intent, “encouraged” as a replacement for “required” would make softer. This would encourage the cultural value of engagement(via voting), without making it a hard requirement.

> All members of a decision-making body are ~~required~~ encouraged to participate in at least 2/3 of formal votes of that decision-making body per calendar year (teams can decide how to account for the specifics of this in low-activity projects, etc.).

Reviewer 2: There are many ways to engage that would be good to recognise and encourage. Observing and giving voice to one’s perspective can be an excellent way to be engaged. For others, doing so may be difficult (and expecting that would put them at a disadvantage).

To inclusively encourage engagement, it would be good to recognize and encourage members to engage in the ways that work best for them, whatever those be.

  - When communications (in whatever number of channels) point to a discussion that requires a project-wide decision, and before major implementation work starts, a standalone issue should be opened that summarizes the issue, points to relevant prior discussions, and calls for an explicit vote/decision. This issue should be tightly scoped to the relevant decision only, and include the `vote` label.
  - When an implementation of a decision appears in a pull request, link to the relevant `vote` issue.
- Decision-making bodies and those proposing decisions should explicitly distinguish between decisions that are two-way doors (easy to reverse later) and one-way doors (difficult or impossible to reverse later). For one-way doors, teams should carefully weigh alternatives and tradeoffs and take extra care to ensure broad participation and stakeholder input. For two-way doors, teams should feel free to move quickly, without compromising the principles and procedures described herein.

## Language for issue/PR template

> Project Jupyter makes decisions using a consensus-seeking model. You can read about this model here (link). The decision making body for this project is documented here (link).

Reviewer: Add a hyphen:
```suggestion
Project Jupyter makes decisions using a consensus-seeking model. You can read about this model here (link). The decision-making body for this project is documented here (link).
```

# Decision Making Guide

This document describes how Jupyter’s governing bodies (Board of Directors, Software Steering Council, Software Projects, Working Groups) make decisions. Because individuals often work across multiple decision making groups, all Jupyter governing bodies are expected to follow this approach.

We seek to honor the principles of collaboration, inclusive participation, and responsive decision making. Some aspects of this decision making guide are required of all teams. Others are provided as recommendations and are optional. We have clearly noted the optional aspects of decision making below.

> Finally, the Jupyter Board of Directors may either intervene, or be called upon by members of a team, in the event that issues arise in the decision making process. This includes, but is not limited to: process ambiguity, violations of the decision making process, mitigating circumstances that require process exceptions, etc.

Author: Today in the governance working group call, we discussed this suggestion and arrived at:

- We should keep the language as is because this document is describing the new governance model.
- This suggestion is to mitigate ambiguity in the transition period, which we should try to keep as brief as possible, but mixing the current SC with the nascent BoD in this document might create confusion.
- We are adding a transition section to the end of this document to help move the process along. That section is designed to be removed once the new governance model is approved.

## Required aspects of decision making

The decision making process is intended to balance broad participation of stakeholders with agility. While the size of decision making bodies may vary (e.g. software projects can range from a handful to dozens of decision makers), all need to find this balance.

All official Jupyter projects, working groups, governing bodies, and official initiatives must have a clearly defined, and documented decision making body. Formally, these decision making bodies make decisions using consensus seeking with an option to call a vote to move the decision forward.

Depending on the governing body, decisions may be proposed by members of the decision making body, or by a broader group of stakeholders/contributors.

> **Informal consensus seeking.** Decision making starts with informal consensus seeking. The goal of this discussion phase is to refine the proposal, consider alternatives, weigh trade-offs, and attempt to find informal consensus. The legitimacy of the consensus seeking process is predicated on all stakeholders having their voices heard; teams must be proactive in providing opportunity for all relevant stakeholders to provide input. If the decision making body arrives at informal consensus, they may immediately move to document and enact the decision. This is the consensus seeking phase.

Reviewer: We had a long conversation about these ideas in the governance office hours today (@afshin, @fperez, @ellisonbg). A few more thoughts out of that conversation:

1. The decision-making body is responsible for actions taken from the "informal consensus" phase in the scope of the project, even if there wasn't a formal vote, so perhaps classifying actions as decisions of the body vs decisions from discussion with the body is more of a distinction without a difference? That still doesn't feel quite right to me that there isn't a difference, but it is a good point that there is still accountability and responsibility even if a vote wasn't taken.
2. If the body is 2 or 3 people, all actively involved in the discussion, taking a vote may be and feel like a pointless formality. The distinction between moving forward with or without a vote becomes more pronounced when the body gets bigger, like five or more. [Roberts Rules of Order in Brief](https://www.amazon.com/Roberts-Rules-Order-Newly-Revised/dp/1541797701) indicates that in-person groups of:
  - 3-4 people can work from consensus discussion
  - as we get to half a dozen or more, we need some sort of formal control, like a chair to maintain who is speaking,  help proposals be worded precisely, and make sure decisions are recorded (and votes are held unless there is obvious total agreement)
  - 12-15 people crosses another threshold, where the meeting becomes essentially "full-scale". To quote, "A certain paradox appears. In order to preserve its freedom to act, the body must impose regulation", not only to keep order, but to ensure that the process is fair to everyone.
3. As we discussed, I think my concerns centered on: without a vote, how do we know we've reached consensus? Perhaps adding a set of guidelines below to help a group signal when consensus is being reached (perhaps depending on the group size) is enough to answer the objection.

> **Informal consensus seeking.** Decision making starts with informal consensus seeking. The goal of this discussion phase is to refine the proposal, consider alternatives, weigh trade-offs, and attempt to find informal consensus. The legitimacy of the consensus seeking process is predicated on all stakeholders having their voices heard; teams must be proactive in providing opportunity for all relevant stakeholders to provide input. If the decision making body arrives at informal consensus, they may immediately move to document and enact the decision. This is the consensus seeking phase.

Author: These comments are addressed in [this commit](https://github.com/jupyter/governance/pull/98/commits/3334e0b6d5929af65ea2e6a4f67956a110a41525). Thanks, Jason!

**Calling a vote.** When the discussion matures during the consensus seeking phase, any member of the decision making body can call the matter to a vote. When that member (the sponsor) calls the vote, they shall summarize the proposal in its current form and notify the entire decision making body. After the proposal is seconded by another member of the decision making body, members have 7 days to vote. The governing body may consider longer voting periods as necessary for special circumstances, or _shorter periods only if all voting members are present_. The result will be determined by a simple majority for binary decisions and ranked choice for multiclass decisions (one among many, or several among many). The sponsor may update the proposal at any point during the voting period, in which case, the voting period will be reset.

> **Voter participation and quorum.** The various Jupyter decision making bodies often have vast scopes and members are often busy. As a result, there are typically a subset of members who have the time, interest, and expertise to dive into the details of a given decision. We believe that open, transparent, and time-bound voting with no quorum will enable these ad-hoc, informal subgroups to form spontaneously on a per-decision basis, in a manner that supports our goals of responsiveness and engagement.

Reviewer 3: ```suggestion
**Voter participation and quorum.** The Jupyter decision making bodies often have broad scope. There are typically a subset of members who have the time, interest, and expertise to dive into the details of a limited scope issue, such as subproject specific issues. We believe that open, transparent, and time-bound voting with no quorum will enable these ad-hoc, informal subgroups to form spontaneously for these limited scope issues and work on a per-decision basis, in a manner that supports our goals of responsiveness and engagement.
```

Reviewer 3: I would remove some of the rationale (busy individuals) that make the governance sound weak. Instead, focusing on the specific governance bounds and intent would improve the clarity. Suggested wording above.

Reviewer 4: Thx Carol, merged!

> **Voter participation and quorum.** The various Jupyter decision making bodies often have vast scopes and members are often busy. As a result, there are typically a subset of members who have the time, interest, and expertise to dive into the details of a given decision. We believe that open, transparent, and time-bound voting with no quorum will enable these ad-hoc, informal subgroups to form spontaneously on a per-decision basis, in a manner that supports our goals of responsiveness and engagement.

Author: 👍  To this change, but we can't commit the suggestion because of syncing issues, so it will be added in a separate commit.

> **Voter participation and quorum.** The various Jupyter decision making bodies often have vast scopes and members are often busy. As a result, there are typically a subset of members who have the time, interest, and expertise to dive into the details of a given decision. We believe that open, transparent, and time-bound voting with no quorum will enable these ad-hoc, informal subgroups to form spontaneously on a per-decision basis, in a manner that supports our goals of responsiveness and engagement.

Reviewer 4: Input was merged, thx!

> All members of a decision making body are required to participate in at least 2/3 of formal votes of that decision making body per calendar year (teams can decide how to account for the specifics of this in low-activity projects, etc.). Members that have not met the 2/3 vote participation threshold for a year will automatically be asked to step down at the end of that year. Those individuals remain eligible to rejoin the decision making body in the future as they become available to participate at the required level. The quorum for all formal votes will be 50% and a "blank" option will always be included, with "blank" option counting towards the quorum but not included in totals for calculating results.

Reviewer 3: I would be explicit on the decision making bodies, including Board of Directors, SSC, and Subprojects (perhaps).

Reviewer 4: Good points folks, thanks! We've added a section at the bottom clarifying these transition points. We made it a separate one so that it can be more easily removed later once the transition is complete, without having to do too much surgery inside the doc.

> All members of a decision-making body are required to participate in at least 2/3 of formal votes of that decision-making body per calendar year (teams can decide how to account for the specifics of this in low-activity projects, etc.). Members that have not met the 2/3 vote participation threshold for a year will automatically be asked to step down at the end of that year. Those individuals remain eligible to rejoin the decision-making body in the future as they become available to participate at the required level. The quorum for all formal votes will be 50% and a "blank" option will always be included, with the "blank" option counting towards the quorum but not included in totals for calculating results.

Reviewer 5: In this paragraph, there is a mention of 50% quorum when in the previous one it says no quorum. I think that is confusing, at a minimum.

Author: Thanks for pointing out the inconsistency, @damianavila! This is resolved now.

> **Recording.** Once a decision has been made during the consensus seeking phase, or by a vote, the decision making body will record the decision in an appropriate location that is as open as possible given the decision and work of the particular body.

Reviewer: ```suggestion
**Recording.** Once a decision has been made during the consensus-seeking phase or by a formal vote, the decision-making body will record the decision in an appropriate location that is as open as possible given the decision and the work of the particular body.
```

## Optional aspects of decision making

While the previous section is prescriptive for all Jupyter teams, the following are recommended guidelines rather than mandatory procedures.

- Do not call a vote to short-circuit an ongoing discussion that is still productive in terms of exploring ideas and feedback. Votes should be called only when discussion has explored the space and stakeholders have provided input.

> - During the consensus seeking phase, don’t interpret a lack of participation, discussion, or feedback as consensus or the lack thereof. Dissent can be silent and sometimes people are supportive but busy.

Reviewer: ```suggestion
- During the consensus-seeking phase, don’t interpret a lack of participation, discussion, or feedback as approval or disapproval. Dissent can be silent and sometimes people are supportive but busy.
```

> - If you are interested in a decision being made, it is your responsibility to encourage voting member/stakeholders/community participation in the decision making process. If you can’t get such participation, you may want to hold off on doing any significant work on the matter.

Reviewer: ```suggestion
- If you are interested in a decision being made, it is your responsibility to encourage voting member/stakeholder/community participation in the decision-making process. If you can’t get such participation, you may want to hold off on doing any significant work on the matter.
```

- Special guidelines for software projects making decisions:

>   - Add a link to this document in your issue and PR templates along with a link to the decision making body membership (see recommended language below).

Reviewer: ```suggestion
  - Add a link to this document in your issue and PR templates along with a link to the decision-making body membership (see recommended language below).
```

>   - Create a “Consensus Seeking” and “Vote” tags to make it easy to find issues/PRs where a decision is being made.

Reviewer: ```suggestion
  - Create “Consensus Seeking” and “Vote” issue tags to make it easy to find issues/PRs where a decision is being made.
```

  - When a vote is called on GitHub, the sponsor should summarize the decision and paste a checklist of the voting members to use in voting in the top entry (description) of the issue/PR.

>   - When communications (in whatever number of channels) point to a discussion that requires a project-wide decision, and before major implementation work starts, a standalone issue should be opened that summarizes the issue, points to relevant prior discussions, and calls for an explicit vote/decision. This issue should be tightly scoped to the relevant decision only, and include the "Vote" tag.

Reviewer 3: It would make sense to have a Vote specific group on Discourse as not all folks monitor all the orgs on GitHub that could possibly contain a Vote.

Reviewer 6: I think this is a good idea to centralize crucial communications in one space. Another option could be to use the `enhancement-proposals` repository.

Whether it's Discourse, a specific repo, a mailing list, etc, there needs to be just a single place that SSC members are **required** to monitor that serves as the "source of truth" for decisions

>   - When an implementation of a decision appears in pull request, link to the relevant “Vote” issue.

Reviewer: ```suggestion
  - When an implementation of a decision appears in a pull request, link to the relevant “Vote” issue.
```

> - Decision making bodies and those proposing decisions should explicitly distinguish between decisions that are two-way doors (easy to reverse later) and one-way doors (difficult or impossible to reverse later). For one-way doors, teams should carefully weigh alternatives and tradeoffs and take extra care to ensure broad participation and stakeholder input. For two-way doors, teams should feel free to move quickly, without compromising the principles and procedures described herein.

Reviewer: ```suggestion
- Decision-making bodies and those proposing decisions should explicitly distinguish between decisions that are two-way doors (easy to reverse later) and one-way doors (difficult or impossible to reverse later). For one-way doors, teams should carefully weigh alternatives and tradeoffs and take extra care to ensure broad participation and stakeholder input. For two-way doors, teams should feel free to move quickly, without compromising the principles and procedures described herein.
```

## Language for issue/PR template

> Project Jupyter makes decisions using a consensus seeking model. You can read about this model here (link). The decision making body for this project is documented here (link).

Reviewer: ```suggestion
Project Jupyter makes decisions using a consensus-seeking model. You can read about this model here (link). The decision making body for this project is documented here (link).
```
